components/core/tests/test-NetworkReader.cpp

```diff
@@ -0,0 +1,173 @@
+#include <cstdint>
+#include <filesystem>
+#include <future>
+#include <memory>
+#include <string>
+#include <string_view>
+#include <thread>
+#include <vector>
+
+#include <Catch2/single_include/catch2/catch.hpp>
+#include <curl/curl.h>
+
+#include "../src/clp/CurlDownloadHandler.hpp"
+#include "../src/clp/ErrorCode.hpp"
+#include "../src/clp/FileReader.hpp"
+#include "../src/clp/NetworkReader.hpp"
+#include "../src/clp/ReaderInterface.hpp"
+
+namespace {
+constexpr size_t cDefaultReaderBufferSize{1024};
+
+[[nodiscard]] auto get_test_input_local_path() -> std::string;
+
+[[nodiscard]] auto get_test_input_remote_url() -> std::string;
+
+[[nodiscard]] auto get_test_input_path_relative_to_tests_dir() -> std::filesystem::path;
+
+/**
+ * @param reader
+ * @param read_buf_size The size of the buffer to use for individual reads from the reader.
+ * @return All data read from the given reader.
+ */
+auto get_content(clp::ReaderInterface& reader, size_t read_buf_size = cDefaultReaderBufferSize)
+        -> std::vector<char>;
+
+auto get_test_input_local_path() -> std::string {
+    std::filesystem::path const current_file_path{__FILE__};
+    auto const tests_dir{current_file_path.parent_path()};
+    return (tests_dir / get_test_input_path_relative_to_tests_dir()).string();
+}
+
+auto get_test_input_remote_url() -> std::string {
+    auto const input_path_relative_to_repo = std::filesystem::path{"components"} / "core" / "tests"
+                                             / get_test_input_path_relative_to_tests_dir();
+    return std::string{"https://raw.githubusercontent.com/y-scope/clp/main/"}
+           + input_path_relative_to_repo.string();
+}
+
+auto get_test_input_path_relative_to_tests_dir() -> std::filesystem::path {
+    return std::filesystem::path{"test_network_reader_src"} / "random.log";
+}
+
+auto get_content(clp::ReaderInterface& reader, size_t read_buf_size) -> std::vector<char> {
+    std::vector<char> buf;
+    // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays)
+    auto const read_buf{std::make_unique<char[]>(read_buf_size)};
+    for (bool has_more_content{true}; has_more_content;) {
+        size_t num_bytes_read{};
+        has_more_content = reader.read(read_buf.get(), read_buf_size, num_bytes_read);
+        std::string_view const view{read_buf.get(), num_bytes_read};
+        buf.insert(buf.cend(), view.cbegin(), view.cend());
+    }
+    return buf;
+}
+}  // namespace
+
+TEST_CASE("network_reader_basic", "[NetworkReader]") {
+    clp::FileReader ref_reader;
+    ref_reader.open(get_test_input_local_path());
+    auto const ref_data{get_content(ref_reader)};
+    ref_reader.close();
+
+    REQUIRE((clp::ErrorCode_Success == clp::NetworkReader::init()));
+    clp::NetworkReader reader(get_test_input_remote_url());
+    auto const streamed_data{get_content(reader)};
+    auto const ret_code{reader.get_curl_return_code()};
+    REQUIRE(ret_code.has_value());
+    // NOLINTNEXTLINE(bugprone-unchecked-optional-access)
```

Comment:
Hm, I don't seem to get this warning.

Reply:
manually running clang-tidy 18.1.1 in terminal leads to this problem